### ISSUE TITLE
Add optional "New" badge for settings sidebar items and display for product research

### ISSUE THREAD

**Issue by quinnkeast** [sourcegraph/sourcegraph#18646]
## Background

From time to time, we make changes to existing features or introduce new features that can be accessed via the user's settings sidebar, and sometimes via the user dropdown menu. Today, we have no way of highlighting these changes, which can make new features hard for users to learn about.

## Proposal

<a href="https://www.figma.com/file/fC9x5nZQep27oWIZY3WJUi/User-settings-sidebar-updates?node-id=275%3A1628" target="_blank"><img width="712" alt="CleanShot 2021-02-25 at 11 26 36@2x" src="https://user-images.githubusercontent.com/6304497/109139937-5dd70000-775c-11eb-9531-a58179bc27f7.png"></a>

<a href="https://www.figma.com/file/fC9x5nZQep27oWIZY3WJUi/User-settings-sidebar-updates?node-id=275%3A1628" target="_blank"><img width="568" alt="CleanShot 2021-02-25 at 11 26 46@2x" src="https://user-images.githubusercontent.com/6304497/109139945-5fa0c380-775c-11eb-8c81-6906612a9430.png"></a>

- [View on Figma](https://www.figma.com/file/fC9x5nZQep27oWIZY3WJUi/User-settings-sidebar-updates?node-id=275%3A1628)

1. [ ] Add an optional "New" badge for menu items in the settings sidebar.
2. [ ] Add an optional "New" badge for menu items in the user dropdown menu.
3. [ ] To support the new "Product research" page, **enable this "New" badge for the "Product research" sidebar menu item**. We should leave this enabled for ~1 month before removing it.

**Comment by valerybugakov** [comment]
@AlicjaSuska @quinnkeast could you give an idea of how urgent is this and what's the expected scope of this task? 
I see three possible implementations:

1. Generic solution that will work for any button/feature in the `settings sidebar` and `user dropdown menu`. It intersects with [Feature Flags RFC](https://docs.google.com/document/d/1aT8uI3mUXpm9IK9_WbXhFM5ahHj9KQeQ521hd9EE5U8/edit#heading=h.trqab8y0kufp) and needs a generic backend solution first.

2. Temporary solution: add new flags for each new badge mentioned in this issue to the site config. Benefits: can be controlled via admin panel.
3. Temporary solution: hard-code these two badges in the source-code and delete them one month later with a separate PR.

The temporary solution should become obsolete once Feature Flags functionality is implemented.
I would go with the third one for now. So we can revisit it later once Feature Flags functionality is ready.

**Comment by quinnkeast** [comment]
Time considerations: I don't want to have multiple "New" badges at the same time. I'd prefer "Product research" be marked as "New" for a period of time, and then "Repositories" once Cloud efforts are ready.

I'd lean towards the hard-coded solution in #3 as well.
